File: week13/25571/donghoony/25571.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

ll T, N, arr[101010];

void solve() {
	cin >> N;
	ll ans = 0;
	for (int i = 0; i < N; i++) cin >> arr[i];
	int l = 0, r = 1;
	while (l < N-1) {
		while (l < N-1 && arr[l] == arr[r]) l++, r++;
		while (r < N-1 && (arr[r] - arr[r-1]) * (arr[r+1] - arr[r]) < 0) r++;
		if (r < N) {
			ll len = (r - l + 1);
			ans += len * (len - 1) / 2;		
		}
		l = r, r++;
	}
	cout << ans << '\n';
}

int main() {
	cin.tie(nullptr)->sync_with_stdio(false);
	cin >> T;
	while (T--) solve();
	return 0;
}

File: week13/25571/donghoony/25571.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    static int readInt() throws IOException {
        return Integer.parseInt(br.readLine());
    }

    static long[] readLongs() throws IOException {
        return Arrays.stream(br.readLine().split(" "))
                .mapToLong(Long::parseLong)
                .toArray();
    }

    public static void main(String[] args) throws IOException {
        int T = readInt();
        for (int i = 0; i < T; i++) solve();
    }

    static void solve() throws IOException {
        int n = readInt();
        long[] arr = readLongs();
        long ans = 0;
        int l = 0, r = 1;
        while (l < n-1) {
            while (l < n-1 && arr[l] == arr[r]) { l++; r++; }
            while (r < n-1 && (arr[r] - arr[r-1]) * (arr[r+1] - arr[r]) < 0) r++;
            long len = r - l + 1;
            if (r < n) ans += len * (len-1) / 2;
            l = r; r++;
        }
        System.out.println(ans);
    }
}
